contracts/Claims.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency, CurrencyLibrary} from "./types/Currency.sol";
+import {IClaims} from "./interfaces/IClaims.sol";
+
+/// An intentionally barebones balance mapping only supporting mint/burn/transfer
+contract Claims is IClaims {
+    using CurrencyLibrary for Currency;
+
+    // Mapping from Currency id to account balances
+    mapping(uint256 => mapping(address => uint256)) public balances;
+
+    /// @inheritdoc IClaims
+    function balanceOf(address account, Currency currency) public view returns (uint256) {
+        return balances[currency.toId()][account];
+    }
+
+    /// @inheritdoc IClaims
+    function transfer(address to, Currency currency, uint256 amount) public {
```

Comment:
Definitely not necessary - but is it weird that we dont have a total supply of each token? Just feels strange that the only way we would have to know how much ETH is held in Claims is by totalling everyones balance lol

Reply:
Yea def will make it hard for accounting but saves 1 SSTORE per mint/burn?

Reply:
yeah it does... just feels weird not to have it 😆 but not against it haha